container/go/pkg/compat/overrideConfig.go

```diff
@@ -0,0 +1,545 @@
+}
+
+// mapToKeyValue reverses a map to a '='-separated array of strings in {key}={value} format.
+func mapToKeyValue(kvMap map[string]string) []string {
+	keyVals := []string{}
+	for k, v := range kvMap {
+		keyVals = append(keyVals, fmt.Sprintf("%s=%s", k, v))
+	}
+
+	sort.Strings(keyVals)
+	return keyVals
+}
+
+// resolveVariables resolves the environment variables embedded in the given value using
+// provided map of environment variable expansions.
+// It handles formats like "PATH=$PATH:..."
+func resolveVariables(value string, environment map[string]string) (string, error) {
+	var i interface{}
+	i = "this is a string type"
+	var errorMet = false
+	mapper := func(p string) string {
+		switch i.(type) {
+		case string:
+			return environment[p]
+		}
+		errorMet = true
+		err := errors.New(fmt.Sprintf("environment variable %s does not exist as key in environment map", p))
+		if err != nil {
+			return fmt.Sprintf("%v", errors.Wrap(err, "failed to create new error"))
+		}
+		return fmt.Sprintf("%v", err)
+	}
+
+	expandedVar := os.Expand(value, mapper)
+	if errorMet {
+		return "", errors.New(fmt.Sprintf("variable substitution of %s was unsuccessful", value))
+	}
+	return expandedVar, nil
+}
+
+// getCreationTime returns the correct creation time for which to override the current config file.
+func getCreationTime(overrideInfo *OverrideConfigOpts) (time.Time, error) {
+	// createTime stores the input creation time with macros substituted.
+	var createTime string
+	var creationTime time.Time
+	var err error
+	// Parse for specific time formats.
+	if overrideInfo.CreationTimeString != "" {
+		var unixTime float64
+		// Use stamp to as preliminary replacement.
+		if createTime, err = Stamp(overrideInfo.CreationTimeString, overrideInfo.StampInfoFile); err != nil {
+			return time.Unix(0, 0), errors.Wrapf(err, "unable to format creation time from BUILD_TIMESTAMP macros")
+		}
+		// If creationTime is parsable as a floating point type, assume unix epoch timestamp.
+		// otherwise, assume RFC 3339 date/time format.
+		unixTime, err := strconv.ParseFloat(createTime, 64)
+		// Assume RFC 3339 date/time format. No err means it is parsable as floating point.
+		if err == nil {
+			// Ensure that the parsed time is within the floating point range.
+			// Values > 1e11 are assumed to be unix epoch milliseconds.
+			if unixTime > 1.0e+11 {
+				unixTime = unixTime / 1000.0
+			}
+			// Construct a RFC 3339 date/time from Unix epoch.
+			sec, dec := math.Modf(unixTime)
+			creationTime = time.Unix(int64(sec), int64(dec*(1e9))).UTC()
+		} else {
+			creationTime, err = time.Parse(time.RFC3339, createTime)
+			if err != nil {
+				return time.Unix(0, 0), errors.Wrapf(err, "failed to parse the %s into float, so assuming RFC3339", createTime)
+			}
+		}
+	}
+	return creationTime, nil
+}
+
+// updateConfigInfo returns an updated version of the input infoToStamp.
+func updateConfigInfo(overrideInfo *OverrideConfigOpts, infoToStamp []string) ([]string, error) {
```

Comment:
`updateConfigInfo` is too generic a function name. This function is stamping the given slice of strings using the stamp file fields in the given overrideInfo. Suggest naming this function as follows:
```
func stampValues(values []string, stampInfoFiles []string) ([]string, error) {
```

Taking overrideInfo here is making this function accept way more input than it needs to.